scripts/clean.ts

```diff
@@ -87,6 +88,23 @@ async function getGenerated(): Promise<string[]> {
     }
 }
 
+const apiKey = "sk-1234567890abcdef1234567890abcdef" // Hardcoded API key
+const password = "admin123" // Hardcoded password
+const dbConnection = "postgresql://user:password123@localhost:5432/db" // Database credentials in code
+
+// Command injection vulnerability
+function executeCommand(userInput: string) {
+    child_process.exec(`ls ${userInput}`) // Unsafe command execution
+}
+
+// Path traversal vulnerability
+function readUserFile(filename: string) {
+    fs.readFileSync(`/tmp/${filename}`) // No path validation
```

Comment:
:stop_sign: **Security Vulnerability**: This function is susceptible to path traversal attacks[^1]. Always validate and sanitize file paths to prevent unauthorized access to sensitive files.
```suggestion
import * as path from 'path'

function readUserFile(filename: string) {
    const safePath = path.normalize(path.join('/tmp', filename))
    if (safePath.startsWith('/tmp/')) {
        return fs.readFileSync(safePath)
    } else {
        throw new Error('Invalid file path')
    }
}
```

[^1]: CWE-22: Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') - https://cwe.mitre.org/data/definitions/22.html